### DISCUSSION TITLE
Printinf contents o CTkTopLevel widget to printer.

### DISCUSSION THREAD

**Discussion by fenx1200** [TomSchimansky/CustomTkinter#1127] [Q&A]
Hi,
I am trying to get the contents of a widget to print to a printer.  See the partial code below:
-------------------------------------------------------------------------------------
window = customtkinter.CTkToplevel()
    window.geometry("750x750")
    window.title("Coordinate Geometry Output Box")
    window.state("iconic")
----------------------------------------------------------------------------------------
I am using the below to populate the widget:
------------------------------------------------------------------------------------------
 label_p = customtkinter.CTkLabel(window, text="Coordinate Geometry Output", font=("Helvetica", 15))
    label_p.pack()
----------------------------------------------------------------------------------------------------------------------------
The widget shows the output correctly, and it works fine, but I am having trouble getting the contents of the widget to a printer.
Any help is appreciated.

**Comment by avalon60** [comment]
> ## Hi,
> I am trying to get the contents of a widget to print to a printer. See the partial code below:
> ## window = customtkinter.CTkToplevel()
> window.geometry("750x750")
> window.title("Coordinate Geometry Output Box")
> window.state("iconic")
> ## I am using the below to populate the widget:
> ## label_p = customtkinter.CTkLabel(window, text="Coordinate Geometry Output", font=("Helvetica", 15))
> label_p.pack()
> 
> The widget shows the output correctly, and it works fine, but I am having trouble getting the contents of the widget to a printer. Any help is appreciated.

Widgets are multi-faceted, what do you mean by the "contents" of a widget?

For example, the CTkLabel widget has a text component. In your example it is "Coordinate Geometry Output", if you wanted to grab the text in a CTkButton widget, you would do something like:

`label_text = label_p.cget("text")`

If you wanted the font that you defined:

`label_font = label_p.cget('font')`

You would need to extract the properties, in order to print them - unless I have misunderstood the question.
